test/integration/plugins/nuagevsp/nuageTestCase.py

```diff
@@ -77,85 +117,104 @@ def setUpClass(cls, zone=None):
         cls.test_data["virtual_machine"]["zoneid"] = cls.zone.id
         cls.test_data["virtual_machine"]["template"] = cls.template.id
 
-        # Create service offering
-        cls.service_offering = ServiceOffering.create(cls.api_client,
-                                                      cls.test_data["service_offering"]
-                                                      )
-        cls._cleanup = [cls.service_offering]
-
         # Check if the host hypervisor type is simulator
-        cls.isSimulator = Hypervisor.list(cls.api_client, zoneid=cls.zone.id)[0].name == "Simulator"
+        cls.isSimulator = Hypervisor.list(
+            cls.api_client,
+            zoneid=cls.zone.id)[0].name == "Simulator"
```

Comment:
If the hypervisor list retrieval fails or returns an empty list, the error returned will be ambiguous.  Please split the retrieval of the list from the indexing operation and asserts that the hypervisor list is not ``None`` and expected length.